cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDataset.java

```diff
@@ -85,12 +91,100 @@ public BusinessMetadataRecord getBusinessMetadata(Id.NamespacedId targetId, Stri
       return null;
     }
 
-    byte[] keyvalue = row.get(KEYVALUE_COLUMN);
     byte[] value = row.get(VALUE_COLUMN);
 
     return new BusinessMetadataRecord(targetId, key, Bytes.toString(value));
   }
 
+  /**
+   * Retrieves the business metadata for the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId the specified {@link Id.NamespacedId}
+   * @return a Map representing the metadata for the specified {@link Id.NamespacedId}
+   */
+  public Map<String, String> getBusinessMetadata(Id.NamespacedId targetId) {
+    String targetType = getTargetType(targetId);
+    MDSKey mdsKey = new MDSKey.Builder().add(targetType).add(targetId.toString()).build();
+    byte[] startKey = mdsKey.getKey();
+    byte[] stopKey = Bytes.stopKeyForPrefix(startKey);
+
+    Map<String, String> metadata = new LinkedHashMap<>();
+    Scanner scan = indexedTable.scan(startKey, stopKey);
+    try {
+      Row next;
+      while ((next = scan.next()) != null) {
+        String key = getMetadataKey(next.getRow());
+        byte[] value = next.get(VALUE_COLUMN);
+        if (key == null || value == null) {
+          continue;
+        }
+        metadata.put(key, Bytes.toString(value));
+      }
+      return metadata;
+    } finally {
+      scan.close();
+    }
+  }
+
+  /**
+   * Removes all business metadata for the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId the {@link Id.NamespacedId} for which metadata is to be removed.
+   */
+  public void removeMetadata(Id.NamespacedId targetId) {
+    removeMetadata(targetId, new Predicate<String>() {
+      @Override
+      public boolean apply(String input) {
+        return true;
+      }
+    });
+  }
+
+  /**
+   * Removes the specified keys from the business metadata of the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId the {@link Id.NamespacedId} for which the specified metadata keys are to be removed.
+   * @param keys the keys to remove from the metadata of the specified {@link Id.NamespacedId}
+   */
+  public void removeMetadata(Id.NamespacedId targetId, final String ... keys) {
+    removeMetadata(targetId, new Predicate<String>() {
+      @Override
+      public boolean apply(String input) {
+        return Arrays.asList(keys).contains(input);
```

Comment:
I think it is better to convert to list outside the predicate, otherwise the array to list conversion may get called for each `apply()` call. Also can we use set instead of list, since contains on a list is `O(n)` operation.